packages/dds/task-manager/src/interfaces.ts

```diff
@@ -19,11 +19,19 @@ export interface ITaskManagerEvents extends ISharedObjectEvents {
 
 export interface ITaskManager extends ISharedObject<ITaskManagerEvents> {
     /**
-     * Try to lock the task.  Promise resolves when the lock is acquired, or rejects if we are removed from the
-     * queue without acquiring the lock for any reason.
+     * Volunteer for the task.  Promise resolves true when the task is assigned to the local client. It rejects if the
+     * local client is removed from the queue without being assigned the task for any reason, such as disconnecting or
+     * abandoning the task while in queue.
      * @param taskId - Identifier for the task
      */
-    lockTask(taskId: string): Promise<void>;
+    volunteerForTask(taskId: string): Promise<boolean>;
+
+    /**
+     * Continuously volunteer to lock the task.  Watch the "assigned" event to determine if the task lock is assigned.
+     * We automatically re-enter the queue if the task lock is lost for any reason.
```

Comment:
Does the client re-enter queue through new op/reuest ? Meaning, it goes to the back of the queue?

Also, what happens if client lock the tasks, and then disconnects without abandoning ? There is no way for other clients to kick the "locking" client out ?

Reply:
> Does the client re-enter queue through new op/reuest ? Meaning, it goes to the back of the queue?

Yes, it will resend a volunteer op (for now at least, I'm considering creating a subscribe op).

> Also, what happens if client lock the tasks, and then disconnects without abandoning ? There is no way for other clients to kick the "locking" client out ?

TaskManager already has a mechanism setup to handle this. To summarize, the quorum will automatically detect when a client disconnects, and there is a listener setup on TaskManager for whenever a member leaves the quorum. At that point the lock will be given to the next client in queue (if the assigned client was the one that left).